src/randomizer.py

```diff
@@ -712,20 +712,22 @@ def find_duplicates(e: Expression) -> None:
     else:
         replace_subexprs(fn.body, find_duplicates)
 
-    assert orig_expr in replace_cands
```

Comment:
worth keeping this assertion I think

Reply:
I had this moved to a different spot, which made sense / was necessary for a different PR I was working on.

But I'll move it back for now, and can discuss moving it for the other PR later